File: flutter_app/lib/models/report.dart
class Report {
  final int? id;
  final String titre;
  final String type;
  final String periode;
  final String? resume;
  final DateTime dateDebut;
  final DateTime dateFin;
  final String? generePar;
  final String? donneesPrincipales;
  final String? conclusions;
  final String? recommandations;
  final String statut;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Report({
    this.id,
    required this.titre,
    required this.type,
    required this.periode,
    this.resume,
    required this.dateDebut,
    required this.dateFin,
    this.generePar,
    this.donneesPrincipales,
    this.conclusions,
    this.recommandations,
    required this.statut,
    this.createdAt,
    this.updatedAt,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'],
      titre: json['titre'] ?? '',
      type: json['type'] ?? '',
      periode: json['periode'] ?? '',
      resume: json['resume'],
      dateDebut: DateTime.parse(json['dateDebut']),
      dateFin: DateTime.parse(json['dateFin']),
      generePar: json['generePar'],
      donneesPrincipales: json['donneesPrincipales'],
      conclusions: json['conclusions'],
      recommandations: json['recommandations'],
      statut: json['statut'] ?? 'BROUILLON',
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'titre': titre,
      'type': type,
      'periode': periode,
      if (resume != null) 'resume': resume,
      'dateDebut': dateDebut.toIso8601String(),
      'dateFin': dateFin.toIso8601String(),
      if (generePar != null) 'generePar': generePar,
      if (donneesPrincipales != null) 'donneesPrincipales': donneesPrincipales,
      if (conclusions != null) 'conclusions': conclusions,
      if (recommandations != null) 'recommandations': recommandations,
      'statut': statut,
    };
  }
}

File: backend/src/main/java/com/healthcare/dashboard/entities/Report.java
package com.healthcare.dashboard.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "reports")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Report {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String titre;
    
    @Column(nullable = false)
    private String type; // FINANCIER, MEDICAL, ACTIVITE, QUALITE, RESSOURCES
    
    @Column(nullable = false)
    private String periode; // JOURNALIER, HEBDOMADAIRE, MENSUEL, ANNUEL
    
    @Column(columnDefinition = "TEXT")
    private String resume;
    
    @Column(nullable = false)
    private LocalDateTime dateDebut;
    
    @Column(nullable = false)
    private LocalDateTime dateFin;
    
    private String generePar;
    
    @Column(columnDefinition = "TEXT")
    private String donneesPrincipales; // JSON format
    
    @Column(columnDefinition = "TEXT")
    private String conclusions;
    
    @Column(columnDefinition = "TEXT")
    private String recommandations;
    
    @Column(nullable = false)
    private String statut; // BROUILLON, PUBLIE, ARCHIVE
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
